### PR TITLE
Ensure all cases have at least one hearing

### PR DESCRIPTION
**Pull request by archferns**
## What

[Link to story](https://dsdmoj.atlassian.net/browse/CACP-456)
1 To ensure we have a HearingSummary present, added a validation.
2. Added a hearing when creating a case via the admin ui
3. Added a migration to fix existing records.

## Checklist

Before you ask people to review this PR:

- [x] Tests and linters should be passing
- [x] Github should not be reporting conflicts; you should have recently run `git rebase master`.
- [x] Avoid mixing whitespace changes with code changes in the same commit. These make diffs harder to read and conflicts more likely.
- [x] You should have looked at the diff against master and ensured that nothing unexpected is included in your changes.
- [x] You should have checked that the commit messages say why the change was made.

### CODE REVIEW

**Comment by jsugarman** [comment]
looks fine to me. but if CP/SIT is able to have hearing summaries with no hearings perhaps we should be allowing representing this behaviour in the mock?!

File: app/models/hearing.rb [hunk 1]
```diff
@@ -3,10 +3,14 @@
...
   include CourtCentreRelatable
+
+  JURISDICTION_TYPES = %w[MAGISTRATES CROWN].freeze
+  LANGUAGES = %w[ENGLISH WELSH].freeze
+
   belongs_to :hearing_type
   belongs_to :cracked_ineffective_trial, optional: true
   has_many :prosecution_case_hearings, dependent: :destroy
-  has_many :prosecution_cases, through: :prosecution_case_hearings
+  has_many :prosecution_cases, through: :prosecution_case_hearings, inverse_of: :hearings
```
Comment (jsugarman) [inline app/models/hearing.rb:13]: can a hearing have more than one case?

Comment (archferns) [reply app/models/hearing.rb:13]: According to the CP structure, yes, however, we haven't encountered anything of that sort in the wild.

**Comment by archferns** [comment]
> looks fine to me. but if CP/SIT is able to have hearing summaries with no hearings perhaps we should be allowing representing this behaviour in the mock?!

HMCTS have confirmed that there will always be at least one hearing (summary) available against prosecution cases.
They are however unable to edit the schema due to internal constraints. Amending the schema will be taken over to the business.

**Pull request merged**